File: HomeBudget.Data/Entities/Bill.cs
namespace HomeBudget.Data.Entities;

public class Bill
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Provider { get; set; } = string.Empty;
    public int DueDay { get; set; }
    public decimal EstimatedAmount { get; set; }
    public string Icon { get; set; } = "receipt";
    public string Color { get; set; } = "#F59E0B";
    public bool IsActive { get; set; } = true;

    public int HouseholdId { get; set; }
    public int CategoryId { get; set; }

    public Household Household { get; set; } = null!;
    public Category Category { get; set; } = null!;
    public ICollection<BillPayment> Payments { get; set; } = new List<BillPayment>();
}

File: HomeBudgetMobile/src/models/bill.ts
import { PaymentMethod, BillStatus } from "./enums";

export interface BillPaymentDto {
  id: number;
  billId: number;
  billName: string;
  amount: number;
  dueDate: string;
  paidDate: string | null;
  paymentMethod: PaymentMethod | null;
  status: BillStatus;
  transactionId: number | null;
}

export interface BillDto {
  id: number;
  name: string;
  provider: string;
  dueDay: number;
  estimatedAmount: number;
  icon: string;
  color: string;
  isActive: boolean;
  categoryName: string;
  categoryId: number;
  nextPayment: BillPaymentDto | null;
  totalPaid: number;
  paymentsCount: number;
  paidThisMonth: boolean;
}

export interface CreateBillDto {
  name: string;
  provider: string;
  dueDay: number;
  estimatedAmount: number;
  icon: string;
  color: string;
  householdId: number;
  categoryId: number;
}

export interface UpdateBillDto {
  id: number;
  name: string;
  provider: string;
  dueDay: number;
  estimatedAmount: number;
  icon: string;
  color: string;
  isActive: boolean;
}

export interface PayBillDto {
  billId: number;
  amount: number;
  paymentMethod: string;
  accountId: number;
  userId: number;
  paidDate: string;
}
